compiler/rustc_const_eval/src/interpret/operand.rs

```diff
@@ -766,13 +774,18 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                             VariantIdx::from_u32(variant_index)
                         } else {
                             dataful_variant
+                        };
+
+                        if relative {
+                            (Scalar::from_uint(variant_index_relative, discr_layout.size), variant)
+                        } else {
+                            // Compute the size of the scalar we need to return.
+                            // No need to cast, because the variant index directly serves as
+                            // discriminant and is encoded in the tag.
```

Comment:
Looking at it here I think this should be before the `if` as it applies to both branches.

Also I am honestly not sure what the first line about the 'size of the scalar' is about. That's just `discr_layout.size`. So I think that first line can be removed (it probably refers to an old version of the code), unless someone else can figure out what it is about. :D